### PR TITLE
Add some more typing to Traffic Routes and save

### PR DESCRIPTION
**Pull request by ViViDboarder**
Adds some more types and makes the saving function more explicit.

Rebased from #529

### CODE REVIEW

**Review by Kane610 — CHANGES_REQUESTED** [review]

**Review by Kane610 — CHANGES_REQUESTED** [review]
The points I marked duplicate, make sure they're sorted alphabetically as well. I think the ones I marked was the duplicate in the wrong order.

**Comment by Kane610** [comment]
Awesome! Only two things left

File: aiounifi/models/traffic_route.py [hunk 1]
```diff
@@ -88,21 +114,59 @@ def id(self) -> str:
...
     @property
     def description(self) -> str:
         """Description given by user to traffic route."""
         return self.raw["description"]
 
+    @property
+    def domains(self) -> list[Domain]:
+        """What IP addresses are matched against by this traffic route.
+
+        Note: This requires clients to use the UniFi network devices as their DNS server.
```
Comment (Kane610) [inline aiounifi/models/traffic_route.py:131]:
```suggestion
        Note: This requires clients to use the UniFi network devices as their DNS server.
```
This line is a bit weird to me, with devices you mean the UniFi gateway/router right?

**Review by Kane610 — CHANGES_REQUESTED** [review]
Final thing Please increase test coverage of your additions to 100%

File: aiounifi/models/traffic_route.py (hunk 1 above)
Comment (ViViDboarder) [reply aiounifi/models/traffic_route.py:131]: I'll reword this to be the exact message from the documentation: `Note: Domain matching requires the client devices to use the UniFi gateway as the DNS server.`

**Comment by ViViDboarder** [comment]
Should be up to 100% now.

File: tests/test_traffic_routes.py [hunk 2]
```diff
@@ -104,8 +110,14 @@ async def test_traffic_routes(unifi_controller, _mock_endpoints, unifi_called_wi
...
     assert traffic_route.description == "TEST 1"
+    assert traffic_route.domains == []
     assert traffic_route.enabled is False
-    assert traffic_route.matching_target == "INTERNET"
+    assert traffic_route.ip_addresses == []
+    assert traffic_route.ip_ranges == []
+    assert traffic_route.matching_target == MatchingTarget.INTERNET
+    assert traffic_route.network_id == "6547e87f6e34654e72293637"
+    assert traffic_route.next_hop == ""
+    assert traffic_route.regions == []
```
Comment (Kane610) [inline tests/test_traffic_routes.py:120]: Since these are very specific it would be good if the data sets where not empty

**Review by Kane610 — CHANGES_REQUESTED** [review]
One final thing

**Comment by ViViDboarder** [comment]
So, this diff ends up changing more. I realized the other test was modifying the fixtures values which munged with other tests. So I made a change to make sure it's a deep copy. That broke other tests so I added some more asserts and made them more robust.

**Review by Kane610 — APPROVED** [review]
Awesome
